pkg/geo/geos/geos.go

```diff
@@ -967,3 +967,46 @@ func Node(a geopb.EWKB) (geopb.EWKB, error) {
 	}
 	return cStringToSafeGoBytes(cEWKB), nil
 }
+
+// VoronoiPolygons Computes the Voronoi Polygons from the vertices of the supplied EWKBs.
+func VoronoiPolygonsWithEnv(a geopb.EWKB, env geopb.EWKB, tolerance float64, onlyEdges int) (geopb.EWKB, error) {
+	g, err := ensureInitInternal()
+	if err != nil {
+		return nil, err
+	}
+	var cEWKB C.CR_GEOS_String
+	if err := statusToError(
+		C.CR_GEOS_VoronoiDiagram(g, goToCSlice(a), goToCSlice(env), C.double(tolerance), C.int(onlyEdges), &cEWKB),
+		); err != nil {
+		return nil, err
+	}
+	return cStringToSafeGoBytes(cEWKB), nil
+}
+
+func VoronoiPolygons(a geopb.EWKB, tolerance float64, onlyEdges int) (geopb.EWKB, error) {
```

Comment:
we should call this VoronoiDiagram -- this library should map to GEOS as directly as possible.